scipy/interpolate/src/_dierckxmodule.cc

```diff
@@ -204,6 +262,247 @@ py_qr_reduce(PyObject* self, PyObject *args, PyObject *kwargs)
 }
 
 
+/*
+ * def _qr_reduce_periodic(double[:, ::1] a, double[:, ::1] a1, double[:, ::1] a2,     # A packed
+                           ssize_t[::1] offset, ssize_t nc, double[:, ::1] y,
+ *                         int k, int64_t len_t, bint init_p=False
+ * ):
+ */
+static PyObject*
+py_qr_reduce_periodic(PyObject* self, PyObject *args, PyObject *kwargs)
+{
+    PyObject *py_a = NULL, *py_h1 = NULL, *py_h2 = NULL, *py_offs = NULL, *py_y = NULL;
+    Py_ssize_t nc;
+    Py_ssize_t py_k;
+    Py_ssize_t len_t;
+    int init_p = false; // Default
+    double p = 0.0;
+    bool get_fp = false; // Default
+    double fp = 0.0;
+    int k;
+
+    // XXX: if the overhead is large, flip back to positional only arguments
+    const char *kwlist[] = {"a", "h1", "h2", "offset",
+                            "nc", "y", "k", "len_t",
+                            "init_p", "get_fp", NULL};
+
+    if(!PyArg_ParseTupleAndKeywords(args, kwargs, "OOOOnOnn|pp", const_cast<char **>(kwlist),
+                                    &py_a, &py_h1, &py_h2, &py_offs, &nc, &py_y, &py_k, &len_t,
+                                    &init_p, &get_fp)) {
```

Comment:
The fact that you need to communicate `get_fp` from C to python is a clear sign that something is not right. Ditto for `init_p`.
Not yet sure what.